onchain/cairo/src/defi/vault.cairo

```diff
@@ -101,23 +135,67 @@ mod Vault {
             ref self: ContractState, token_address: ContractAddress, amount: u256
         ) {
             let caller = get_caller_address();
-        // Check if token valid
-
-        // Receive/burn token minted
-
-        // Resend amount of coin deposit by user
-
+            // Check if token valid
+            assert(self.is_token_permitted(token_address), 'Non permited token');
+
+            // Receive/burn token minted
+            let token_mintable = IERC20MintableDispatcher {
+                contract_address: self.token_address.read()
+            };
+            token_mintable.burn(caller, amount);
+
+            // Resend amount of coin deposit by user
+            let token_deposited = IERC20Dispatcher { contract_address: token_address };
+            let amount_ratio = amount / self.token_permitted.read(token_address).ratio_mint;
```

Comment:
Please comment with a TODO, and just add the amount without the ratio at the moment